Assets/LeapMotionModules/Interaction/InteractionHand.cs

```diff
@@ -76,26 +86,73 @@ private struct HoverCheckResults {
+      Array.Clear(tempPerFingerHovered, 0, tempPerFingerHovered.Length);
 
       HoverCheckResults results = new HoverCheckResults() {
         hovered = _hoverableCache,
+        perFingerHovered = tempPerFingerHovered,
         primaryHovered = null,
         primaryHoveredDistance = float.PositiveInfinity,
         checkedHand = hand
       };
 
-      foreach (var interactionObj in hoverCandidates) {
-        if (interactionObj.ignoreHover) continue;
-        CheckHoverForElement(hand, interactionObj, ref results);
+      //Loop through all the fingers (that we care about)
+      if (hand != null) {
+        float leastHandDistance = float.PositiveInfinity;
+        for (int i = 0; i < 3; i++) {
+          if (!hand.Fingers[i].IsExtended) { continue; }
+          float leastFingerDistance = float.PositiveInfinity;
+          Vector3 fingerTip = hand.Fingers[i].TipPosition.ToVector3();
+
+          //Loop through all the candidates
+          foreach (InteractionBehaviourBase elem in hoverCandidates) {
+            if (elem.ignoreHover) continue;
+            ISpaceComponent element = elem.GetComponent<ISpaceComponent>();
+            if (element != null) {
+              CheckHoverForElement(fingerTip, elem, element, i, leastFingerDistance, leastHandDistance, ref results);
+            } else {
+              CheckHoverForBehavior(fingerTip, elem, ref results);
+            }
+          }
+        }
       }
 
       return results;
     }
 
-    private void CheckHoverForElement(Hand hand, InteractionBehaviourBase hoverable, ref HoverCheckResults curResults) {
-      float distance = hoverable.GetDistance(_intentionPosition);
+    public Vector3 transformPoint(Vector3 worldPoint, ISpaceComponent element) {
+      Vector3 localPos = element.anchor.space.transform.InverseTransformPoint(worldPoint);
+      return element.anchor.space.transform.TransformPoint(element.anchor.transformer.InverseTransformPoint(localPos));
+    }
+
+    public void coarseInverseTransformHand(Hand inHand, ISpaceComponent element) {
+      Vector3 localPalmPos = element.anchor.space.transform.InverseTransformPoint(inHand.PalmPosition.ToVector3());
+      Quaternion localPalmRot = element.anchor.space.transform.InverseTransformRotation(inHand.Rotation.ToQuaternion());
+
+      inHand.SetTransform(element.anchor.space.transform.TransformPoint(element.anchor.transformer.InverseTransformPoint(localPalmPos)),
+                          element.anchor.space.transform.TransformRotation(element.anchor.transformer.InverseTransformRotation(localPalmPos, localPalmRot)));
+    }
+
+    private void CheckHoverForElement(Vector3 position, InteractionBehaviourBase behaviour, ISpaceComponent element, int whichFinger, float leastFingerDistance, float leastHandDistance, ref HoverCheckResults curResults) {
+      //NEED BETTER DISTANCE FUNCTION
+      float dist = Vector3.SqrMagnitude(((Component)element).transform.position - transformPoint(position, element));
```

Comment:
this function needs to match hoverable.GetDistance or else you might get weird comparisons between objects that are inside a curved space vs objects that are not